readme.md

```diff
@@ -1 +1,258 @@
-# Functional Programming
\ No newline at end of file
+# Functional Programming
+
+## Learning Objectives
+
+- Identify the benefits of pure functions and avoiding side effects
+- Define immutability and how it relates to pure functions
+- Explain what recursion is and why we use it
+- Explain the concepts of first class and higher order functions
+- Use `filter`, `map`, `reduce`, `sort`
+
+## Framing
+
+### Why functional programming? Why now?
+
```

Comment:
Let's talk a little bit about paradigms in programming....

Could turn it into a quick STWG - "What are some examples of paradigms we've followed so far in this class?"